tests/fixture/testnet/local/network.go

```diff
@@ -0,0 +1,644 @@
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	cfg "github.com/ava-labs/avalanchego/config"
+	"github.com/ava-labs/avalanchego/genesis"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/tests/fixture/testnet"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ava-labs/avalanchego/utils/perms"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+// This interval was chosen to avoid spamming node APIs during
+// startup, as smaller intervals (e.g. 50ms) seemed to noticeably
+// increase the time for a network's nodes to be seen as healthy.
+const networkHealthCheckInterval = 200 * time.Millisecond
+
+var (
+	errInvalidNodeCount        = errors.New("failed to populate local network config: non-zero node count is only valid for a network without nodes")
+	errInvalidKeyCount         = errors.New("failed to populate local network config: non-zero key count is only valid for a network without keys")
+	errLocalNetworkDirNotSet   = errors.New("local network directory not set - has Create() been called?")
+	errInvalidNetworkDir       = errors.New("failed to write local network: invalid network directory")
+	errNotHealthyBeforeTimeout = errors.New("failed to see all nodes healthy before timeout")
+)
+
+// Default root dir for storing networks and their configuration.
+func GetDefaultRootDir() (string, error) {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(homeDir, ".testnetctl", "networks"), nil
+}
+
+// Find the next available network ID by attempting to create a
+// directory numbered from 1000 until creation succeeds. Returns the
+// network id and the full path of the created directory.
+func FindNextNetworkID(rootDir string) (uint32, string, error) {
+	var (
+		networkID uint32 = 1000
+		dirPath   string
+	)
+	for {
+		_, reserved := constants.NetworkIDToNetworkName[networkID]
+		if reserved {
+			continue
+		}
+
+		dirPath = filepath.Join(rootDir, fmt.Sprint(networkID))
+		err := os.Mkdir(dirPath, perms.ReadWriteExecute)
+		if err == nil {
+			return networkID, dirPath, nil
+		}
+
+		if !errors.Is(err, fs.ErrExist) {
+			return 0, "", fmt.Errorf("failed to create network directory: %w", err)
+		}
+
+		// Directory already exists, keep iterating
+		networkID++
+	}
+}
+
+// Defines the configuration required for a local network (i.e. one composed of local processes).
+type LocalNetwork struct {
+	testnet.NetworkConfig
+	LocalConfig
+
+	// Nodes with local configuration
+	Nodes []*LocalNode
+
+	// Path where network configuration will be stored
+	Dir string
+}
+
+// Returns the configuration of the network in backend-agnostic form.
+func (ln *LocalNetwork) GetConfig() testnet.NetworkConfig {
+	return ln.NetworkConfig
+}
+
+// Returns the nodes of the network in backend-agnostic form.
+func (ln *LocalNetwork) GetNodes() []testnet.Node {
+	nodes := []testnet.Node{}
```

Comment:
Interesting... I would have thought that this would fail linting... But it seems that the linter isn't that smart. Had this been written as:
`var nodes []testnet.Node`
It would have told you to pre-allocate the slice like:
`nodes := make([]testnet.Node, 0, len(ln.Nodes))`